qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -301,6 +301,22 @@ def _remove_tab(self, tab):
 
     def undo(self):
         """Undo removing of a tab."""
+        # Remove unused tab which may be created after the last tab is closed
+        last_close = config.get('tabs', 'last-close')
+        if last_close in ['blank', 'startpage', 'default-page']:
+            only_one_tab_open = self.count() == 1
+            no_history = self.widget(0).history().count() == 1
+            urls = {
+                'blank': QUrl('about:blank'),
+                'startpage': QUrl(config.get('general', 'startpage')[0]),
+                'default-page': config.get('general', 'default-page'),
+            }
+            first_tab_url = self.widget(0).url()
```

Comment:
I wonder if `self.widget(0).page().mainFrame().requestedUrl()` would work better for redirects etc. here - can you try?